wrappers/services_test.go

```diff
@@ -260,18 +267,29 @@ Before=slices.target
 X-Snappy=yes
 
 [Slice]
+# Always enable cpu accounting, so the following cpu quota options have an effect
+CPUAccounting=true
+CPUQuota=%[2]d%%
+AllowedCPUs=%[3]s
+
 # Always enable memory accounting otherwise the MemoryMax setting does nothing.
 MemoryAccounting=true
-MemoryMax=%[2]s
+MemoryMax=%[4]d
 # for compatibility with older versions of systemd
-MemoryLimit=%[2]s
+MemoryLimit=%[4]d
 
 # Always enable task accounting in order to be able to count the processes/
 # threads, etc for a slice
 TasksAccounting=true
+TasksMax=%[5]d
 `
 
-	sliceContent := fmt.Sprintf(sliceTempl, grp.Name, memLimit.String())
+	allowedCpusValue := strings.Trim(strings.Join(strings.Fields(fmt.Sprint(resourceLimits.CPU.AllowedCPUs)), ","), "[]")
```

Comment:
That's a bit hard to comprehend, what triggered me to write this:
```

func sliceAsStringStrings(vals []int) string {
	return strings.Trim(strings.Join(strings.Fields(fmt.Sprint(vals)), ","), "[]")
}

func sliceAsStringBuilder(vals []int) string {
	b := &strings.Builder{}
	for i, v := range vals {
		fmt.Fprintf(b, "%d", v)
		if i != len(vals)-1 {
			fmt.Fprintf(b, ",")
		}
	}
	return b.String()
}

var allowedCPUs = []int{1, 2, 3, 4, 5, 6, 7, 8}

func BenchmarkJoinWithStrings(b *testing.B) {
	for i := 0; i < b.N; i++ {
		sliceAsStringStrings(allowedCPUs)
	}
}

func BenchmarkJoinWithBuilder(b *testing.B) {
	for i := 0; i < b.N; i++ {
		sliceAsStringBuilder(allowedCPUs)
	}
}
```
yielding:
```
OK: 43 passed
goos: linux
goarch: amd64
pkg: github.com/snapcore/snapd/strutil
cpu: AMD Ryzen 7 1700 Eight-Core Processor          
BenchmarkJoinWithStrings-16       406503              2889 ns/op
BenchmarkJoinWithBuilder-16       599824              2014 ns/op
PASS
ok      github.com/snapcore/snapd/strutil       2.499s
```

Reply:
Made it event faster :smile: 
```
BenchmarkJoinWithStrings-16       392541              2902 ns/op
BenchmarkJoinWithBuilder-16       768690              1571 ns/op
```

Reply:
Check this out (8x faster):
```
BenchmarkJoinWithStrings-16      2115771              2889 ns/op
BenchmarkJoinWithBuilder-16     18142676               326.8 ns/op
```
with the code:
```
func sliceAsStringBuilder(vals []int) string {
	b := &strings.Builder{}
	last := len(vals) - 1
	for i, v := range vals {
		b.WriteString(strconv.Itoa(v))
		if i != last {
			b.WriteRune(',')
		}
	}
	return b.String()
}
```

Reply:
Really nice speed improvements, and defintely much easier to read! Ill get that added